components/nexus-repository/src/main/java/org/sonatype/nexus/repository/group/GroupHandler.java

```diff
@@ -69,40 +72,82 @@ public String toString() {
-    DispatchedRepositories dispatched = request.getAttributes().getOrCreate(DispatchedRepositories.class);
-
-    Repository repository = context.getRepository();
-    GroupFacet group = repository.facet(GroupFacet.class);
-
-    // FIXME: To move forward prototype, exposing HTTP semantics here
-    String method = request.getAction();
+    final String method = context.getRequest().getAction();
     switch (method) {
       case GET:
       case HEAD: {
-        for (Repository member : group.members()) {
-          log.trace("Trying member: {}", member);
-
-          // track repositories we have dispatched to, prevent circular dispatch for nested groups
-          if (dispatched.contains(member)) {
-            log.trace("Skipping already dispatched member: {}", member);
-            continue;
-          }
-          dispatched.add(member);
-
-          ViewFacet view = member.facet(ViewFacet.class);
-          Response response = view.dispatch(request);
-          if (response.getStatus().getCode() == HttpStatus.OK) {
-            return response;
-          }
-        }
-        return HttpResponses.notFound();
+        final DispatchedRepositories dispatched = context.getRequest().getAttributes()
+            .getOrCreate(DispatchedRepositories.class);
+        return doGet(context, dispatched);
       }
 
       default:
         return HttpResponses.methodNotAllowed(method, GET, HEAD);
     }
   }
+
+  /**
+   * Method that actually performs group GET. Override if needed.
+   */
+  protected Response doGet(final @Nonnull Context context,
+                           final @Nonnull DispatchedRepositories dispatched)
+      throws Exception
+  {
+    final GroupFacet groupFacet = context.getRepository().facet(GroupFacet.class);
+    return getFirst(context.getRequest(), groupFacet.members(), dispatched);
+  }
+
+  /**
+   * Returns the first OK response from member repositories or {@link HttpResponses#notFound()} if none of the members
+   * responded with OK.
+   */
+  protected Response getFirst(final @Nonnull Request request,
+                              final @Nonnull List<Repository> members,
+                              final @Nonnull DispatchedRepositories dispatched)
+      throws Exception
+  {
+    for (Repository member : members) {
+      log.trace("Trying member: {}", member);
+      // track repositories we have dispatched to, prevent circular dispatch for nested groups
+      if (dispatched.contains(member)) {
+        log.trace("Skipping already dispatched member: {}", member);
+        continue;
+      }
+      dispatched.add(member);
+
+      final ViewFacet view = member.facet(ViewFacet.class);
+      final Response response = view.dispatch(request);
+      if (response.getStatus().getCode() == HttpStatus.OK) {
+        return response;
+      }
+    }
+    return HttpResponses.notFound();
+  }
+
+  /**
+   * Returns all responses from all members as a linked map, where order is processing order (actually group member
```

Comment:
can skip the 'actually' bit here; if the order is group member order just say so ;)